src/main/java/org/isf/accounting/gui/PatientBillEdit.java

```diff
@@ -915,38 +1005,27 @@ private JScrollPane getJScrollPaneBalance() {
 
 	private JTable getJTableBalance() {
 		if (jTableBalance == null) {
-			jTableBalance = new JTable();
-			jTableBalance.setModel(new DefaultTableModel(new Object[][] {
-					{
-							"<html><b>"+MessageBundle.getMessage("angal.newbill.balance.txt").toUpperCase()+"</b></html>",
-							currencyCod,
-							balance}
-			},
-					new String[] {"","",""}) {
-				private static final long serialVersionUID = 1L;
-				Class<?>[] types = new Class<?>[] { JLabel.class, JLabel.class, Double.class, };
-
-				@Override
-				public Class<?> getColumnClass(int columnIndex) {
-					return types[columnIndex];
-				}
-
-				@Override
-				public boolean isCellEditable(int row, int column) {
-					return false;
-				}
-			});
-			jTableBalance.getColumnModel().getColumn(1).setMinWidth(CURRENCY_CODE_WIDTH);
-			jTableBalance.getColumnModel().getColumn(1).setMaxWidth(CURRENCY_CODE_WIDTH);
-			jTableBalance.getColumnModel().getColumn(2).setMinWidth(PRICE_WIDTH);
-			jTableBalance.getColumnModel().getColumn(2).setMaxWidth(PRICE_WIDTH);
-			jTableBalance.setMaximumSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
-			jTableBalance.setMinimumSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
-			jTableBalance.setPreferredSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
+			jTableBalance = new JTable(new JTableBalanceModel());
+			sizeJTableBalance();
 		}
 		return jTableBalance;
 	}
 
+	private void setJTableBalance() {
+		jTableBalance.setModel(new JTableBalanceModel());
+		sizeJTableBalance();
+	}
+
+	private void sizeJTableBalance() {
+		jTableBalance.getColumnModel().getColumn(1).setMinWidth(CURRENCY_CODE_WIDTH);
+		jTableBalance.getColumnModel().getColumn(1).setMaxWidth(CURRENCY_CODE_WIDTH);
+		jTableBalance.getColumnModel().getColumn(2).setMinWidth(PRICE_WIDTH);
+		jTableBalance.getColumnModel().getColumn(2).setMaxWidth(PRICE_WIDTH);
+		jTableBalance.setMaximumSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
```

Comment:
wouldn't it be better to create the Dimension object once and refer to it 3 times?